test/e2e/kubelet_stats.go

```diff
@@ -287,6 +352,120 @@ func formatResourceUsageStats(nodeName string, containerStats map[string]*contai
 	return fmt.Sprintf("Resource usage on node %q:\n%s", nodeName, buf.String())
 }
 
+type int64arr []int64
+
+func (a int64arr) Len() int           { return len(a) }
+func (a int64arr) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a int64arr) Less(i, j int) bool { return a[i] < a[j] }
+
+type usageDataPerContainer struct {
+	cpuData        []float64
+	memUseData     []int64
+	memWorkSetData []int64
+}
+
+func computePercentiles(timeSeries map[time.Time]resourceUsagePerContainer, percentilesToCompute []int) map[int]resourceUsagePerContainer {
+	if len(timeSeries) == 0 {
+		return make(map[int]resourceUsagePerContainer)
+	}
+	dataMap := make(map[string]*usageDataPerContainer)
+	for _, v := range timeSeries {
+		for k, _ := range v {
+			dataMap[k] = &usageDataPerContainer{
+				cpuData:        make([]float64, 0),
+				memUseData:     make([]int64, 0),
+				memWorkSetData: make([]int64, 0),
+			}
+		}
+		break
+	}
+	for _, singleStatistic := range timeSeries {
+		for name, data := range singleStatistic {
+			dataMap[name].cpuData = append(dataMap[name].cpuData, data.CPUUsageInCores)
+			dataMap[name].memUseData = append(dataMap[name].memUseData, data.MemoryUsageInBytes)
+			dataMap[name].memWorkSetData = append(dataMap[name].memWorkSetData, data.MemoryWorkingSetInBytes)
+		}
+	}
+	for _, v := range dataMap {
+		sort.Float64s(v.cpuData)
+		sort.Sort(int64arr(v.memUseData))
+		sort.Sort(int64arr(v.memWorkSetData))
+	}
+
+	result := make(map[int]resourceUsagePerContainer)
+	for _, perc := range percentilesToCompute {
+		data := make(resourceUsagePerContainer)
+		for k, v := range dataMap {
+			dataSize := len(v.cpuData)
+			data[k] = &containerResourceUsage{
+				Name:                    k,
+				CPUUsageInCores:         v.cpuData[int(math.Ceil(float64(dataSize*perc)/100))-1],
```

Comment:
please refactor "int(math.Ceil(float64(dataSize*perc)/100))-1" to a separate var and use it